core/src/main/java/io/grpc/internal/SpiffeIdParser.java

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright 2024 The gRPC Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.grpc.internal;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import com.google.common.base.Splitter;
+import java.util.Locale;
+
+/**
+ * Parses URI to SPIFFE ID if it matches SPIFFE ID standard.
+ * <p>
+ * @see <a href="https://github.com/spiffe/spiffe/blob/master/standards/SPIFFE-ID.md">standard</a>
+ */
+public final class SpiffeIdParser {
+
+  private static final String PREFIX = "spiffe://";
+
+  private SpiffeIdParser() {}
```

Comment:
optional/nit: For a non-instantiable class, generally this is the last line of the class.

Example: https://github.com/grpc/grpc-java/blob/master/netty/src/main/java/io/grpc/netty/Utils.java

Reply:
It exists in both variations in the codebase, let's ask Eric -
https://github.com/grpc/grpc-java/blob/1dae144f0a3013e8056adf9c78eb095a1dc8fcad/core/src/main/java/io/grpc/internal/JsonParser.java#L40

Reply:
I found the third style (no constructor) - https://github.com/grpc/grpc-java/blob/master/core/src/main/java/io/grpc/internal/JsonUtil.java so I’m dropping it here. Even if user instantiates the class using a default one, I don’t think they can misuse the static APIs

Reply:
We do want the private constructor for utility classes that just hold statics. That's more of an Effective Java thing.